pandas/plotting/_matplotlib/tools.py

```diff
@@ -289,6 +289,22 @@ def _remove_labels_from_axis(axis):
     axis.get_label().set_visible(False)
 
 
+# If two axes sharing an axis also have the same position, they can be referred to as
+# internally sharing the common axis (a.k.a twinning). _handle_shared_axes() is only
+# interested in axes externally sharing an axis, regardless of whether either of the
+# axes is also internally sharing with a third axes.
```

Comment:
could this go into a docstring for _ignore_colocated_axes?

Reply:
Done